ament_ikos/ament_ikos/main.py

```diff
@@ -64,19 +174,37 @@ def main(argv=sys.argv[1:]) -> int:
         'directory',
         default=os.curdir,
         help='Directory to scan recursively for ikos marker files.')
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate an xunit-compliant XML file')
+    parser.add_argument(
+        '--sarif-file',
+        help='Generate a SARIF-compliant output file')
     args = parser.parse_args(argv)
 
     ikos_db_files = []
+
+    # Scan for the marker files left by IKOS (*.ikosbin) in the target directory
     marker_files = scan_marker_files(args.directory)
     if not marker_files:
         print('No marker files found when scanning ' + args.directory)
         return 0
 
+    # Process each one
     for m in marker_files:
-        result = process_marker_file(m)
+        result = process_marker_file(m, args)
         if result:
             ikos_db_files.append(result)
 
+    # Generate the output files
+    test_name = f'{os.path.basename(args.directory)}.ikos'
+    if args.xunit_file:
+        aggregate_junit_xml_files(ikos_db_files, args.xunit_file, test_name)
+    if args.xunit_file:
```

Comment:
Should this be `sarif_file` or is the aggregate file only used when junit is specified?

```suggestion
    if args.sarif_file:
```